### PR TITLE
Added metrics baseline API

### PR DESCRIPTION
**Pull request by asafst**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5623 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5623/commits/8b3f26fe2450bd6e7952f08138c817091fb9c46f

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2017-11-01-preview/baseline_API.json [hunk 1]
```diff
@@ -41,7 +41,7 @@
           "tags": [
             "Baseline"
           ],
-          "operationId": "MetricBaseline_Get",
+          "operationId": "Baseline_Get",
```
Comment (veronicagg) [inline specification/monitor/resource-manager/Microsoft.Insights/preview/2017-11-01-preview/baseline_API.json]: operationId updates will cause breaking changes in existing SDKs, are you intentionally making this change? The operationIds in the specification correspond to the name and grouping of the methods when generating SDK code. 
See https://github.com/Azure/azure-sdk-for-python/pull/4714/files#diff-ca529633360fbb2a6253fb399e08cf29R18 as an example

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2019-03-01/metricBaselines_API.json [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+          ],
+          "responses": {
+            "default": {
+              "description": "Error response describing why the operation failed.",
+              "schema": {
+                "$ref": "#/definitions/ErrorResponse"
+              }
+            },
+            "200": {
+              "description": "Successful request to get the list of metric values.",
```
Comment (veronicagg) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2019-03-01/metricBaselines_API.json]: is the operation returning a list?

**Review by veronicagg — CHANGES_REQUESTED** [review]

File: specification/monitor/resource-manager/Microsoft.Insights/preview/2017-11-01-preview/baseline_API.json (hunk 1 above)
Comment (asafst) [reply specification/monitor/resource-manager/Microsoft.Insights/preview/2017-11-01-preview/baseline_API.json]: It caused a conflict in the build of the Java SDK, but I reverted the breaking change and updated the new API

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2019-03-01/metricBaselines_API.json (hunk 2 above)
Comment (asafst) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2019-03-01/metricBaselines_API.json]: Yes, I updated the operation to indicate this is a List operation. I also updated some descrptions to indicate this.

Comment (veronicagg) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2019-03-01/metricBaselines_API.json]: Ok, the response model doesn't look like a list, is that the proper response? Should the operation be x-ms-pageable? https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-pageable

**Comment by veronicagg** [comment]
@asafst have you been able to review this with the ARM team? Not sure if you're connecting offline, I didn't see any posts from them here.

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2019-03-01/metricBaselines_API.json [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+            "description": "Error code",
+            "type": "string"
+          },
+          "message": {
+            "description": "Error message indicating why the operation failed.",
+            "type": "string"
+          }
+        }
+      },
+      "MetricBaselinesResponse": {
```
Comment (KrisBash) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2019-03-01/metricBaselines_API.json]: I understand that this is trying to align to /metrics API, but this response violates the Resource Provider contract in its current form. 
1) The URI: /metricBaselines implies a collection and should return a collection response (value: [{....}]).  If it is a true singleton, you can use /metricBaselines/default, but the collection path should return the collection response in that case as well.
2) The response is not well formed, and should have the format:
`{"id":"<id>",
"type":"<type">,
"name":"<name>",
"properties":{}
}`

In effect, most of this should be pushed into the "properties" object. 

While this deviates from the response for /metrics, it achieves contractual compliance, enables response authorization checks, and establishes consistency with the remainder of APIs.

Comment (KrisBash) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2019-03-01/metricBaselines_API.json]: You can use references to the common proxy resource model in https://github.com/Azure/azure-rest-api-specs/blob/master/specification/common-types/resource-management/v1/types.json

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2019-03-01/metricBaselines_API.json [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+          "sensitivity": {
+            "type": "string",
+            "enum": [
+              "Low",
+              "Medium",
+              "High"
+            ],
+            "x-ms-enum": {
+              "name": "Sensitivity",
+              "modelAsString": false
```
Comment (KrisBash) [inline specification/monitor/resource-manager/Microsoft.Insights/stable/2019-03-01/metricBaselines_API.json]: Recommend modelAsString: true. It allows the addition of enum values without breaking the clients

**Review by KrisBash — CHANGES_REQUESTED** [review]
Changes requested

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2019-03-01/metricBaselines_API.json (hunk 3 above)
Comment (asafst) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2019-03-01/metricBaselines_API.json]: Ok, this makes sense. While seperating this API even more from the /metrics API, the common structure makes sense. I've updated the swagger to return a collection (only value[{}]) and each item in the correct format as above

File: specification/monitor/resource-manager/Microsoft.Insights/stable/2019-03-01/metricBaselines_API.json (hunk 4 above)
Comment (asafst) [reply specification/monitor/resource-manager/Microsoft.Insights/stable/2019-03-01/metricBaselines_API.json]: great suggestion, thanks. updated to true

**Review by KrisBash — APPROVED** [review]
LGTM - thanks for the updates

**Comment by asafst** [comment]
@veronicagg this PR was approved by ARM, I would apreciate if you could take another look. Thanks!

**Review by veronicagg — APPROVED** [review]

**Comment by veronicagg** [comment]
@asafst sorry about the delay merging, thought I had clicked the button.

**Pull request merged**